### DISCUSSION TITLE
c61d8b7cf7 crashed in tr_peerMgrPeerStats

### DISCUSSION THREAD

**Discussion by dantearmok** [transmission/transmission#5517] [Support] [answered]
Mac mini (late 2012, 2.5-GHz 4-Core i5), 16GB RAM, plus an assortment of USB & FW HDs.
macOS 10.15 Catalina, PeakHour, PIA, and Xcode 12.3 (all fully updated).

Running Transmission-c61d8b7cf7 Build #9060 (May 9, 2023, 1:29:27 AM)

Transmission was running in background, during screen saver...

Process:               Transmission [3452]
Path:                  /Users/USER/*/Transmission.app/Contents/MacOS/Transmission
Identifier:            Transmission
Version:               4.1.0-dev (14714.4.1.0)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Transmission [3452]
User ID:               502

Date/Time:             2023-05-12 05:27:25.725 -0400
OS Version:            Mac OS X 10.15.7 (19H2026)
Report Version:        12
Anonymous UUID:        9D83084F-22F2-20FB-B0D2-056E94905F2F

Time Awake Since Boot: 97000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [3452]

VM Regions Near 0:
--> 
    __TEXT                 000000010ab7b000-000000010adb7000 [ 2288K] r-x/r-x SM=COW  /Users/USER/*/Transmission.app/Contents/MacOS/Transmission

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   ???                           	000000000000000000 0 + 0
1   org.m0k.transmission          	0x000000010ac89ad4 tr_peerMgrPeerStats(tr_torrent const*, unsigned long*) + 311 (peer-mgr.cc:1643)
2   org.m0k.transmission          	0x000000010abbdf39 -[Torrent peers] + 59 (Torrent.mm:943)
3   org.m0k.transmission          	0x000000010ac10ac5 -[InfoPeersViewController updateInfo] + 1060 (InfoPeersViewController.mm:154)
4   org.m0k.transmission          	0x000000010abd7514 -[InfoWindowController updateInfoStats] + 48 (InfoWindowController.mm:470)
5   org.m0k.transmission          	0x000000010ab8d965 -[Controller updateUI] + 922 (Controller.mm:2318)
6   com.apple.Foundation          	0x00007fff3515ad6a __NSFireTimer + 67
7   com.apple.CoreFoundation      	0x00007fff32a55718 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
8   com.apple.CoreFoundation      	0x00007fff32a5527e __CFRunLoopDoTimer + 872
9   com.apple.CoreFoundation      	0x00007fff32a54d69 __CFRunLoopDoTimers + 322
10  com.apple.CoreFoundation      	0x00007fff32a399ed __CFRunLoopRun + 1885
11  com.apple.CoreFoundation      	0x00007fff32a38c33 CFRunLoopRunSpecific + 466
12  com.apple.HIToolbox           	0x00007fff31654aad RunCurrentEventLoopInMode + 292
13  com.apple.HIToolbox           	0x00007fff316547c5 ReceiveNextEventCommon + 584
14  com.apple.HIToolbox           	0x00007fff31654569 _BlockUntilNextEventMatchingListInModeWithFilter + 64
15  com.apple.AppKit              	0x00007fff2fc9d3c9 _DPSNextEvent + 883
16  com.apple.AppKit              	0x00007fff2fc9bc10 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
17  com.apple.AppKit              	0x00007fff2fc8d91e -[NSApplication run] + 658
18  com.apple.AppKit              	0x00007fff2fc5f706 NSApplicationMain + 777
19  libdyld.dylib                 	0x00007fff6c7bbcc9 start + 1

Thread 1:
0   org.m0k.transmission          	0x000000010ac8ba9a tr_peerMgr::makeNewPeerConnections(unsigned long) + 676 (peer-mgr.cc:2488)
1   org.m0k.transmission          	0x000000010ac8b5ef tr_peerMgr::reconnectPulse() + 2851
2   org.m0k.transmission          	0x000000010ac8ca6e tr_peerMgr::bandwidthPulse() + 214 (peer-mgr.cc:2222)
3   org.m0k.transmission          	0x000000010ad1afdb event_process_active_single_queue + 1121
4   org.m0k.transmission          	0x000000010ad17e7b event_base_loop + 1134 (event.c:2006)
5   org.m0k.transmission          	0x000000010ac78585 tr_session_thread_impl::sessionThreadFunc(event_base*) + 87 (session-thread.cc:250)
6   org.m0k.transmission          	0x000000010ac78977 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> >(void*) + 63 (thread:300)
7   libsystem_pthread.dylib       	0x00007fff6c9c4109 _pthread_start + 148
8   libsystem_pthread.dylib       	0x00007fff6c9bfb8b thread_start + 15

Thread 2:
0   libsystem_kernel.dylib        	0x00007fff6c8ff746 __semwait_signal + 10
1   libsystem_c.dylib             	0x00007fff6c882eea nanosleep + 196
2   libc++.1.dylib                	0x00007fff69acb6da std::__1::this_thread::sleep_for(std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > const&) + 73
3   org.m0k.transmission          	0x000000010acb4360 tr_web::Impl::curlThreadFunc() + 448
4   org.m0k.transmission          	0x000000010acb45d4 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (tr_web::Impl::*)(), tr_web::Impl*> >(void*) + 59 (thread:300)
5   libsystem_pthread.dylib       	0x00007fff6c9c4109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6c9bfb8b thread_start + 15

Thread 3:
0   libsystem_kernel.dylib        	0x00007fff6c901756 kevent + 10
1   org.m0k.transmission          	0x000000010ac2416c -[VDKQueue watcherThread:] + 203 (VDKQueue.mm:190)
2   com.apple.Foundation          	0x00007fff350f4222 __NSThread__start__ + 1064
3   libsystem_pthread.dylib       	0x00007fff6c9c4109 _pthread_start + 148
4   libsystem_pthread.dylib       	0x00007fff6c9bfb8b thread_start + 15

Thread 4:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	0x00007fff6c8fcdea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6c8fd160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff32a3ad07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff32a397c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff32a38c33 CFRunLoopRunSpecific + 466
5   com.apple.AppKit              	0x00007fff2fe3ebf4 _NSEventThread + 132
6   libsystem_pthread.dylib       	0x00007fff6c9c4109 _pthread_start + 148
7   libsystem_pthread.dylib       	0x00007fff6c9bfb8b thread_start + 15

Thread 5:: com.apple.coreaudio.AQClient
0   libsystem_kernel.dylib        	0x00007fff6c8fcdea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6c8fd160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff32a3ad07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff32a397c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff32a38c33 CFRunLoopRunSpecific + 466
5   com.apple.audio.toolbox.AudioToolbox	0x00007fff30daff60 GenericRunLoopThread::Entry(void*) + 166
6   libAudioToolboxUtility.dylib  	0x00007fff69143611 CAPThread::Entry(CAPThread*) + 77
7   libsystem_pthread.dylib       	0x00007fff6c9c4109 _pthread_start + 148
8   libsystem_pthread.dylib       	0x00007fff6c9bfb8b thread_start + 15

Thread 6:
0   libsystem_kernel.dylib        	0x00007fff6c8fce26 semaphore_wait_trap + 10
1   com.apple.audio.caulk         	0x00007fff662ffb16 caulk::mach::semaphore::wait() + 16
2   com.apple.audio.caulk         	0x00007fff662ff9b2 caulk::semaphore::timed_wait(double) + 106
3   com.apple.audio.caulk         	0x00007fff662ff7c4 caulk::concurrent::details::worker_thread::run() + 30
4   com.apple.audio.caulk         	0x00007fff662ff1e4 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 45
5   libsystem_pthread.dylib       	0x00007fff6c9c4109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6c9bfb8b thread_start + 15

Thread 7:: AMCP Logging Spool
0   libsystem_kernel.dylib        	0x00007fff6c8fce26 semaphore_wait_trap + 10
1   com.apple.audio.caulk         	0x00007fff662ffb16 caulk::mach::semaphore::wait() + 16
2   com.apple.audio.caulk         	0x00007fff662ff9b2 caulk::semaphore::timed_wait(double) + 106
3   com.apple.audio.caulk         	0x00007fff662ff7c4 caulk::concurrent::details::worker_thread::run() + 30
4   com.apple.audio.caulk         	0x00007fff662ff1e4 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 45
5   libsystem_pthread.dylib       	0x00007fff6c9c4109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6c9bfb8b thread_start + 15

Thread 8:
0   libsystem_pthread.dylib       	0x00007fff6c9bfb68 start_wqthread + 0

Thread 9:
0   libsystem_pthread.dylib       	0x00007fff6c9bfb68 start_wqthread + 0

Thread 10:
0   libsystem_pthread.dylib       	0x00007fff6c9bfb68 start_wqthread + 0

Thread 11:
0   libsystem_pthread.dylib       	0x00007fff6c9bfb68 start_wqthread + 0

Thread 12:
0   libsystem_pthread.dylib       	0x00007fff6c9bfb68 start_wqthread + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x000000010adc7f78  rbx: 0x00006000038f4de8  rcx: 0x0000000000000010  rdx: 0xffffffffffffffd0
  rdi: 0x00007ffee5081ed0  rsi: 0x00007fa4e48a2c00  rbp: 0x00007ffee5082000  rsp: 0x00007ffee5081ec8
   r8: 0x000000000000001b   r9: 0x000000000000002f  r10: 0x00007fa4e4a42400  r11: 0xffffffa5fd546b10
  r12: 0x00000000645e0641  r13: 0x00007fa4de93d580  r14: 0x00007fa4e25c8ae0  r15: 0x00007fa4e48a2c00
  rip: 0x0000000000000000  rfl: 0x0000000000010287  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000014 (no mapping for user instruction read)
Trap Number:     14

**Comment by dantearmok** [comment]
Had another crash, same ilk
tr_peerMgrPeerStats(tr_torrent const*, unsigned long*) + 311 (peer-mgr.cc:1643)

Have since updated to tonite's nightly.  So far so stable.

fwiw,
- Dan.

**Comment by ckerr** [comment, marked as answer]
Converted to issue in https://github.com/transmission/transmission/issues/5524
